megamek/src/megamek/client/ui/swing/DeployMinefieldDisplay.java

```diff
@@ -140,9 +140,7 @@ private void beginMyTurn() {
         setActiveEnabled(p.getNbrMFActive());
         setInfernoEnabled(p.getNbrMFInferno());
         setRemoveMineEnabled(true);
-        if (!p.hasMinefields()) {
-            butDone.setEnabled(true);
-        }
+        butDone.setEnabled(true);
```

Comment:
Do we even need to set it to enabled then? Do we ever disable it?

Reply:
Yes, when it's not the current player's turn.